lib/kitchen_sink/map.ex

```diff
@@ -56,7 +56,9 @@ defmodule KitchenSink.Map do
     right = clean_struct(right)
     do_deep_merge(left, right)
   end
-
+  # When the left argument is a struct, ensure that its own keys are the only
```

Comment:
Kept multiple variations of the same function clustered together https://github.com/planswell/kitchen-sink/pull/40/commits/32228069a9657f424553375e0f8b95ae63ca3841.